src/Sindarin/SindarinDebugger.class.st

```diff
@@ -70,14 +70,13 @@ SindarinDebugger >> contextIsAboutToSignalException: aContext [
 
 	| node |
 	node := self node.
-	node isMessage ifFalse: [ ^ false ].
+	self isMessageSend ifFalse: [ ^ false ].
 	(#( #signal #signalIn: ) includes: node selector) ifFalse: [ ^ false ].
 	aContext basicSize >= 1 ifFalse: [ ^ false ].
 	(Exception allSubclasses includes: (aContext at: aContext basicSize))
 		ifTrue: [ ^ true ]. "#signal sent to a subclass of Exception"
-	(Exception allSubclasses includes:
-		 (aContext at: aContext basicSize) class) ifTrue: [ ^ true ]. "#signal sent to an instance of a subclass of Exception"
-	^ false
+	^(Exception allSubclasses includes: (aContext at: aContext basicSize) class) "#signal sent to an instance of a subclass of Exception"
```

Comment:
Cannot it be simplified with `isKindOf:` ?

It's still ugly but ...

Reply:
Perhaps.
It is ugly and it should be simplified.
This is also part of the original code.